src/base/OpenMCCellAverageProblem.C

```diff
@@ -1708,16 +1708,28 @@ OpenMCCellAverageProblem::initializeTallies()
       _current_mean_tally.resize(n_translations);
       _previous_mean_tally.resize(n_translations);
 
-      // create a new mesh; by setting the ID to -1, OpenMC will automatically detect the
-      // next available ID
-      auto mesh = std::make_unique<openmc::LibMesh>(_mesh_template_filename, _scaling);
-      mesh->set_id(-1);
-      mesh->output_ = false;
+      std::unique_ptr<openmc::LibMesh> local_mesh; // local mesh that will be created from a libMesh pointer or a mesh file
+      if(_mesh_template_filename.empty())
+      {
+        // create a local mesh from a libMesh mesh
+        local_mesh = std::make_unique<openmc::LibMesh>(_mesh.getMesh(),_scaling);
+        // by setting the ID to -1, OpenMC will automatically detect the next available ID
+        local_mesh->set_id(-1);
```

Comment:
We can move the `local_mesh->set_id` and `local_mesh->output_` lines outside of the if-else statement since they're identical between the two different mesh approaches.

Reply:
Moved these outside